apis/iam/v1beta1/diff_test.go

```diff
@@ -0,0 +1,856 @@
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+					{
+						Role:    "roles/editor",
+						Members: []Member{"user:editor@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID:  "spec.bindings[role=roles/editor]",
+						Old: "present in desired spec",
+						New: "absent in actual spec",
+					},
+				},
+			},
+		},
+		{
+			name: "binding removed",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+					{
+						Role:    "roles/editor",
+						Members: []Member{"user:editor@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID:  "spec.bindings[role=roles/editor]",
+						Old: "absent in desired spec",
+						New: "present in actual spec",
+					},
+				},
+			},
+		},
+		{
+			name: "binding members differ",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com", "user:other@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.bindings[role=roles/viewer]",
+						Old: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:other@example.com", "user:viewer@example.com"},
+						},
+						New: IAMPolicyBinding{
+							Role:    "roles/viewer",
+							Members: []Member{"user:viewer@example.com"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "binding members order differs - no diff",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com", "user:other@example.com"},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:other@example.com", "user:viewer@example.com"},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{},
+		},
+		{
+			name: "binding condition differs",
+			desired: &IAMPolicySpec{
+				Bindings: []IAMPolicyBinding{
+					{
+						Role:    "roles/viewer",
+						Members: []Member{"user:viewer@example.com"},
```

Comment:
The test case `binding members differ` could be more explicit in its naming, for example `binding members differ - member removed`.